evalml/utils/gen_utils.py

```diff
@@ -355,6 +355,26 @@ def _convert_woodwork_types_wrapper(pd_data):
     return pd_data
 
 
+def _to_woodwork(X, y, to_pandas=True):
```

Comment:
A bit confused about this method, especially since it's called `_to_woodwork` but can also return pandas? I'm guessing there are two things that are being done here: conversion to WW and back, and the check for all_numeric. But I wonder if these two should be mingled together in a util method for generic purposes like this when it seems pretty specific to the data splitter (allowing only numeric)? I remember wondering if it was worth creating a util method that converts to WW and then back but then decided not realllly since in some cases, we want to do some checks/grab some information (here, the numeric cols and whether all cols are numeric).

Long ramble of thoughts but I guess my suggestion here would be, could we at least rename this to something more descriptive :p

Reply:
Makes sense. I wanted to make it a util to reduce repetitive code, but now that @chukarsten helped me whittle down the code a lot more, I guess it isn't as necessary for this one. I've left it as a method in `gen_utils` and changed the name to something hopefully more useful, but also open to just having it as a method in the base classes for the data splitters.